lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -0,0 +1,408 @@
+                    500
+                )  # When we upgrade the disk too the resize progress takes some more time.
+            if not self.module.check_mode:
+                self.hcloud_server.change_type(
+                    server_type=self.client.server_types.get_by_name(server_type),
+                    upgrade_disk=self.module.params.get("upgrade_disk"),
+                ).wait_until_finished(timeout)
+                if (
+                    state == "present"
+                    and previous_server_status == Server.STATUS_RUNNING
+                ) or state == "started":
+                    self.start_server()
+
+            self._mark_as_changed()
+        self._get_server()
+
+    def start_server(self):
+        if self.hcloud_server.status != Server.STATUS_RUNNING:
+            if not self.module.check_mode:
+                self.client.servers.power_on(self.hcloud_server).wait_until_finished()
+            self._mark_as_changed()
+        self._get_server()
+
+    def stop_server(self):
+        if self.hcloud_server.status != Server.STATUS_OFF:
+            if not self.module.check_mode:
+                self.client.servers.power_off(self.hcloud_server).wait_until_finished()
+            self._mark_as_changed()
+        self._get_server()
+
+    def present_server(self):
+        self._get_server()
+        if self.hcloud_server is None:
+            self._create_server()
+        else:
+            self._update_server()
+
+    def delete_server(self):
+        self._get_server()
+        if self.hcloud_server is not None:
+            if not self.module.check_mode:
+                self.client.servers.delete(self.hcloud_server).wait_until_finished()
+            self._mark_as_changed()
+        self.hcloud_server = None
+
+    @staticmethod
+    def define_module():
+        return AnsibleModule(
+            argument_spec=dict(
+                id={"type": "int"},
+                name={"type": "str"},
+                image={"type": "str"},
+                server_type={"type": "str"},
+                location={"type": "str"},
+                datacenter={"type": "str"},
+                user_data={"type": "str"},
+                ssh_keys={"type": "list", "aliases": ["ssh_key"]},
+                volumes={"type": "list"},
+                backups={"type": "bool", "default": False},
+                upgrade_disk={"type": "bool", "default": False},
+                force_upgrade={"type": "bool", "default": False},
+                state={
+                    "choices": ["present", "absent", "restarted", "started", "stopped"],
+                    "default": "present",
+                },
+                **Hcloud.base_module_arguments()
+            ),
+            supports_check_mode=True,
+        )
+
+
+def main():
+    module = AnsibleHcloudServer.define_module()
+
+    hcloud = AnsibleHcloudServer(module)
+    state = module.params.get("state")
+    if state == "absent":
+        hcloud.delete_server()
+    else:
```

Comment:
Why this additional indentation and not just a long if-elif-elif-elif-else chain ?